CMakeLists.txt

```diff
@@ -851,6 +860,32 @@ if(BOINC)
 	endif()
 endif()
 
+if(LIBARCHIVE)
+	add_custom_command(OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/libArchive_Fwrapper.c POST_BUILD COMMAND ${CMAKE_COMMAND} -E copy ${CMAKE_SOURCE_DIR}/SixTrack/libArchive_Fwrapper.c ${CMAKE_BINARY_DIR}/libArchive_Fwrapper.c DEPENDS copy_input_files)
+	add_custom_command(OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/libArchive_wrapper.c POST_BUILD COMMAND ${CMAKE_COMMAND} -E copy ${CMAKE_SOURCE_DIR}/SixTrack/libArchive_wrapper.c ${CMAKE_BINARY_DIR}/libArchive_wrapper.c DEPENDS copy_input_files)
+	add_custom_command(OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/libArchive_wrapper.h POST_BUILD COMMAND ${CMAKE_COMMAND} -E copy ${CMAKE_SOURCE_DIR}/SixTrack/libArchive_wrapper.h ${CMAKE_BINARY_DIR}/libArchive_wrapper.h DEPENDS copy_input_files)
+	
+	include_directories(${LIBARCHIVE_SOURCE_DIR}/libarchive)
+	
+	add_library(libarchive STATIC IMPORTED)
+	if(WIN32)
+		set_target_properties(libarchive PROPERTIES IMPORTED_LOCATION ${LIBARCHIVE_BUILD_DIR}/libarchive/libarchive_static.a)
+		
+		find_library(pthread STATIC pthread)
+		find_library(bz2 STATIC bz2)
+		find_library(Bcrypt STATIC Bcrypt)
+		find_library(iconv STATIC iconv)
+		target_link_libraries(SixTrack libarchive bz2 Bcrypt pthread iconv)
+	else()
+		set_target_properties(libarchive PROPERTIES IMPORTED_LOCATION ${LIBARCHIVE_BUILD_DIR}/libarchive/libarchive.a)
+		target_link_libraries(SixTrack libarchive)
+
+		find_library(pthread STATIC pthread)
+		find_library(z STATIC z)
+		target_link_libraries(SixTrack z pthread)
```

Comment:
I'm needing this there:

                find_library(bz2 STATIC bz2)
                target_link_libraries(SixTrack bz2 z pthread)